src/pybind/mgr/volumes/fs/volume.py

```diff
@@ -422,6 +422,37 @@ def remove_subvolume(self, fs_handle, **kwargs):
             ret = self.volume_exception_to_retval(ve)
         return ret
 
+    @connection_pool_wrap
+    def resize_subvolume(self, fs_handle, **kwargs):
+        ret        = 0, "", ""
+        subvolname = kwargs['sub_name']
+        newsize    = kwargs['new_size']
+        groupname  = kwargs['group_name']
+        noshrink   = kwargs['no_shrink']
+
+        try:
+            with SubVolume(self.mgr, fs_handle) as sv:
+                spec = SubvolumeSpec(subvolname, groupname)
+                if not self.group_exists(sv, spec):
+                    raise VolumeException(
+                        -errno.ENOENT, "Subvolume group '{0}' not found, create it with " \
+                        "'ceph fs subvolumegroup create' before creating or resizing subvolumes".format(groupname))
+                subvolpath = sv.get_subvolume_path(spec)
+                if not subvolpath:
+                    raise VolumeException(
+                        -errno.ENOENT, "Subvolume '{0}' not found, create it with " \
+                        "'ceph fs subvolume create' before resizing subvolumes".format(subvolname))
+                nsize, usedbytes = sv.resize_subvolume(subvolpath, newsize, noshrink)
+                if nsize:
+                    percentage = '{0:.2f}'.format((float(usedbytes) / nsize) * 100.0)
+                else:
+                    percentage = '{0:.2f}'.format(float(usedbytes) * 100.0) # as nsize tends to 0
```

Comment:
This doesn't look right. Quota=0, means nsize is 'infinity', right? Anyway, the percentage should be undefined. Can't we just set `percent = 'undefined'`?